<documentation title="Superfluous PHPDoc">
    <standard>
        <![CDATA[
    Function must only have PHPDoc if it either has at least one untyped parameter, returns non void or throws an
    exception.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: All required annotations are specified.">
            <![CDATA[
/**
 * @param float $bar
 * @return string
 * @throws BarIsNotAnArrayException
 */
public function getFoo($bar)
{
    if (!is_float($bar)) {
        throw new BarIsNotAnArrayException();
    }

    return 'foo';
}
        ]]>
        </code>
        <code title="Valid: Abstract method returns void.">
            <![CDATA[
/**
 * @return void
 */
abstract protected function prepareFoo();
        ]]>
        </code>
        <code title="Valid: Interface method returns void.">
            <![CDATA[
/**
 * @return void
 */
public function prepareFoo();
        ]]>
        </code>
        <code title="Invalid: Superfluous PHPDoc.">
            <![CDATA[
/**
 * @return void
 */
public function prepareFoo()
{
    $this->foo = 'foo';
}
        ]]>
        </code>
        <code title="Invalid: Function has only typed parameters.">
            <![CDATA[
/**
 * @param Foo $foo
 */
public function prepareFoo(Foo $foo)
{
    $this->foo = $foo;
}
        ]]>
        </code>
    </code_comparison>
</documentation>
